tests/cggi_to_tfhe_rust/add_one.mlir

```diff
@@ -6,33 +6,79 @@
-!ct_ty = !lwe.lwe_ciphertext<encoding = #encoding>
-!pt_ty = !lwe.lwe_plaintext<encoding = #encoding>
-func.func @add_one(%arg0: !ct_ty, %arg1: !ct_ty, %arg2: !ct_ty, %arg3: !ct_ty, %arg4: !ct_ty, %arg5: !ct_ty, %arg6: !ct_ty, %arg7: !ct_ty) -> (!ct_ty, !ct_ty, !ct_ty, !ct_ty, !ct_ty, !ct_ty, !ct_ty, !ct_ty) {
-  %false = arith.constant false
-  %0 = lwe.encode %false {encoding = #encoding} : i1 to !pt_ty
-  %1 = lwe.trivial_encrypt %0 : !pt_ty to !ct_ty
-  %2 = cggi.lut3(%arg0, %arg1, %1) {lookup_table = 6 : ui8} : !ct_ty
-  %3 = lwe.encode %false {encoding = #encoding} : i1 to !pt_ty
-  %4 = lwe.trivial_encrypt %3 : !pt_ty to !ct_ty
-  %5 = lwe.encode %false {encoding = #encoding} : i1 to !pt_ty
-  %6 = lwe.trivial_encrypt %5 : !pt_ty to !ct_ty
-  %7 = cggi.lut3(%arg0, %4, %6) {lookup_table = 1 : ui8} : !ct_ty
-  %8 = cggi.lut3(%arg0, %arg1, %arg2) {lookup_table = 120 : ui8} : !ct_ty
-  %9 = cggi.lut3(%arg0, %arg1, %arg2) {lookup_table = 128 : ui8} : !ct_ty
-  %10 = lwe.encode %false {encoding = #encoding} : i1 to !pt_ty
-  %11 = lwe.trivial_encrypt %10 : !pt_ty to !ct_ty
-  %12 = cggi.lut3(%9, %arg3, %11) {lookup_table = 6 : ui8} : !ct_ty
-  %13 = cggi.lut3(%9, %arg3, %arg4) {lookup_table = 120 : ui8} : !ct_ty
-  %14 = cggi.lut3(%9, %arg3, %arg4) {lookup_table = 128 : ui8} : !ct_ty
-  %15 = lwe.encode %false {encoding = #encoding} : i1 to !pt_ty
-  %16 = lwe.trivial_encrypt %15 : !pt_ty to !ct_ty
-  %17 = cggi.lut3(%14, %arg5, %16) {lookup_table = 6 : ui8} : !ct_ty
-  %18 = cggi.lut3(%14, %arg5, %arg6) {lookup_table = 120 : ui8} : !ct_ty
-  %19 = cggi.lut3(%14, %arg5, %arg6) {lookup_table = 128 : ui8} : !ct_ty
-  %20 = lwe.encode %false {encoding = #encoding} : i1 to !pt_ty
-  %21 = lwe.trivial_encrypt %20 : !pt_ty to !ct_ty
-  %22 = cggi.lut3(%19, %arg7, %21) {lookup_table = 6 : ui8} : !ct_ty
-  return %22, %18, %17, %13, %12, %8, %2, %7 : !ct_ty, !ct_ty, !ct_ty, !ct_ty, !ct_ty, !ct_ty, !ct_ty, !ct_ty
+// CHECK: [[CONCAT:%.*]] = tensor.from_elements
+// CHECK: return [[CONCAT]] : tensor<8x!tfhe_rust.eui3>
+
+module {
+  func.func @add_one(%arg0: tensor<8x!lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>>) -> tensor<8x!lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>> {
+    %true = arith.constant true
+    %false = arith.constant false
+    %c7 = arith.constant 7 : index
+    %c6 = arith.constant 6 : index
+    %c5 = arith.constant 5 : index
+    %c4 = arith.constant 4 : index
+    %c3 = arith.constant 3 : index
+    %c2 = arith.constant 2 : index
+    %c1 = arith.constant 1 : index
+    %c0 = arith.constant 0 : index
+    %extracted = tensor.extract %arg0[%c0] : tensor<8x!lwe.lwe_ciphertext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>>
+    %0 = lwe.encode %true {encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>} : i1 to !lwe.lwe_plaintext<encoding = #lwe.unspecified_bit_field_encoding<cleartext_bitwidth = 3>>
```

Comment:
would it be too onerous for readability sake to add a type alias like `!plaintext_ty` and `!ciphertext_ty` so move these long types to the top?

Reply:
Done! These testcases are just the intermediate IRs printed from the TF2FHE pipeline, so I don't pay too much attention to their format.

At some point, I was trying to see if there's some easy way to get the IR printer to always use type aliases, but I couldn't find an easy way to do that. Let me know if you ever stumble across it!